src/components/SearchField/SearchField.jsx

```diff
@@ -0,0 +1,137 @@
+import _ from 'lodash';
+import React, { createElement } from 'react';
+import { lucidClassNames } from '../../util/style-helpers';
+import { createClass, findTypes, omitProps } from '../../util/component-types';
+
+import TextField from '../TextField/TextField';
+import SearchIcon from '../Icon/SearchIcon/SearchIcon';
+
+import reducers from './SearchField.reducers';
+
+const cx = lucidClassNames.bind('&-SearchField');
+
+const {
+	bool,
+	func,
+	node,
+	number,
+	oneOfType,
+	string,
+} = React.PropTypes;
+
+/**
+* {"categories": ["controls", "text"], "madeFrom": ["TextField", "SearchIcon"]}
+*
+* This is a container that renders panels and controls its expansion/retraction.
+*/
+const SearchField = createClass({
+	displayName: 'SearchField',
+
+	components: {
+		TextField,
+		Icon: createClass({
+			displayName: 'SearchField.Icon',
+			propName: 'Icon',
+		}),
+	},
+
+	reducers,
+
+	propTypes: {
```

Comment:
should wait until it's requested, right now there's a way a consumer can accomplish the same effect, if we notice a lot repeat usage like that, we can make it a new prop